File: solutions/scala/darts/1/src/main/scala/Darts.scala
object Darts:
   def score(x: Double, y: Double): Int =
      val r = math.sqrt(x * x + y * y)
      r match
         case a if a > 5 & a <= 10 => 1
         case b if b > 1 & b <= 5  => 5
         case c if c <= 1          => 10
         case _                    => 0

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

import Data.List (find)

score :: Float -> Float -> Int
score x y = maybe 0 snd $ find (\(r, _) -> d <= r) ranges
  where
    d = sqrt (x * x + y * y)
    ranges = [(1, 10), (5, 5), (10, 1)]